chat-extensions-compose/src/main/kotlin/com/ably/chat/extensions/compose/Presence.kt

```diff
@@ -0,0 +1,43 @@
+package com.ably.chat.extensions.compose
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import com.ably.chat.PresenceMember
+import com.ably.chat.Room
+import com.ably.chat.RoomStatus.Attached
+import com.ably.chat.annotations.ExperimentalChatApi
+import com.ably.chat.asFlow
+import kotlinx.coroutines.cancelAndJoin
+import kotlinx.coroutines.launch
+
+/**
+ * @return currently present members
+ */
+@ExperimentalChatApi
+@Composable
+public fun Room.collectAsPresenceMembers(): List<PresenceMember> {
```

Comment:
I
```suggestion
public fun Room.collectAsPresenceMembers(): List<PresenceMember> {
```
the method returns current presence members, so is it possible to rename it that way. 
Not sure if there's specific method syntax for `@ Composable` methods

Reply:
like `Room.getCurrentPresenceMembers()`
Same goes for other methods

Reply:
`get` is a very generic name, and it can also conflict with property getters and setters. Extension functions that transform an observable state into composable data usually include this information in their name. See https://developer.android.com/develop/ui/compose/state#use-other-types-of-state-in-jetpack-compose.

All Android extensions follow one of these naming patterns:
	•	collectAs
	•	subscribeAs
	•	observeAs

I chose collectAs because, in these composable extensions, we are mostly collecting a Flow.

Reply:
Regarding `currentPresenceMembers`, I get the impression that "presence" already implies currently present members. @AndyTWF wdyt?

Reply:
I agree, "presence" in its own right applies "whoever is currently present".

OOI, do we want to move these properties to the respective classes, e.g. `Presence.collectAsMembers`, rather than having it at the `Room` level?

Reply:
he main reason is that having them at the room level makes it possible to use the room’s status internally. I don’t want to introduce a backlink to the room in the public API of a specific "feature". I also don’t think it looks ugly or misleading.

Additionally, it’s more consistent to have a custom extension that combines data from two different "features" —something like `CurrentlyTypingPresenceMember`. I’m not sure if it’s useful, but who knows? :)

Reply:
Ah, can we not pass down the `RoomStatus` object to avoid using the rooms API internally?

Reply:
We can, but I think it will look weird :)